### PR TITLE
Add docker compose file and documentation

### PR DESCRIPTION
**Pull request by stfung77**
Add docker compose file and documentation

docker compose file will spin up a working setup to test against (Anchor Platform, Reference Server, Kafka, and a Postgres Database)

### CODE REVIEW

**Review by marcelosalloum — COMMENTED** [review]
I wonder if we can reuse some existing yaml files instead of creating new ones that will need to be maintained in the future.

Maybe we could update the `loadConfigYaml` method to be able to take a yaml that _"inherits"_ from another one.

https://github.com/stellar/java-stellar-anchor-sdk/blob/265ade0ae2a9f75c4cdaa1715630e08f87519a50/platform/src/main/java/org/stellar/anchor/platform/configurator/PropertiesReader.java#L79-L91

If you prefer to address that in another PR, could you plz create an issue for that?

File: anchor-reference-server/src/main/resources/anchor-reference-server-docker-compose-config.yaml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+# Spring Data JDBC settings for Postgres
+#
+#spring.jpa.generate-ddl: true
+#spring.jpa.database: POSTGRESQL
+#spring.jpa.show-sql: false
+#spring.datasource.driver-class-name: org.postgresql.Driver
+#spring.datasource.url: jdbc:postgresql://host.docker.internal:5432/anchorplatform
+#spring.datasource.username: anchorplatform
+#spring.datasource.password: <PASSWORD>
+#spring.mvc.converters.preferred-json-mapper: gson
```
[the remark this answers is not in the document]
Comment (stfung77) [reply anchor-reference-server/src/main/resources/anchor-reference-server-docker-compose-config.yaml:74]: h2 is used on line 47 which i think is ok in this case

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      - ./platform/src/main/resources:/config
+      - ./anchor-reference-server/src/main/resources:/reference-config
+    ports:
+      - "8080:8080"
+      - "8081:8081"
+
+#  db:
+#    image: postgres:latest
+#    ports:
+#      - "5432:5432"
```
[the remark this answers is not in the document]
Comment (stfung77) [reply docker-compose.yaml:24]: i was thinking postgres would be optional when running with docker compose. any thoughts on this? not sure why but depends_on - db still works though.

File: platform/src/main/resources/anchor-docker-compose-config.yaml [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+    #
+    # If the `type` is `data-spring-jdbc`, Spring Data JDBC will be used to read the application data.
+    #
+    # `data-spring-jdbc` is the only `type` value supported for now.
+    #
+    data-access:
+      type: data-spring-jdbc
+      settings: data-spring-jdbc-sqlite           # The absolute location of the configuration data in this yaml file
+      #settings: data-spring-jdbc-local-postgres    # use local postgres instance
+
```
[the remark this answers is not in the document]
Comment (stfung77) [reply platform/src/main/resources/anchor-docker-compose-config.yaml]: added postgres default for docker compose config

**Pull request merged**